src/SignupWithMobile.js

```diff
@@ -157,11 +159,27 @@ export default class SignupWithMobile extends React.PureComponent<
   };
 
   onPressSubmitButton = () => {
-    if (this.props.onPressSubmitButton) {
-      this.props.onPressSubmitButton({
-        countryCallingCode: this.state.countryCallingCode,
-        nationalNumber: this.state.nationalNumber,
-      });
+    const { countryCallingCode, nationalNumber } = this.state;
+    try {
+      const phoneNumber = parsePhoneNumber(
+        `${countryCallingCode} ${nationalNumber}`
```

Comment:
It is better to pass the phone number as `+${countryCallingCode}${nationalNumber}` so that it is unambiguous.